deluge/component.py

```diff
@@ -210,10 +217,18 @@ def on_pause(result):
     def _component_resume(self):
         def on_resume(result):
             self._component_state = 'Started'
+            if need_start:
```

Comment:
I was thinking about preventing the next lines from being called in case we already called them...
For example, in case we are in `_component_resume` and need to start the timer. If the component has the attribute `resume`, we need to call `resume()` and to `on_resume()` as a callback, in which we will start the timer.
But, if the component don't this attribute, I call `_component_start_timer` and only than `on_resume()`.
In this situation, I'll be calling `_component_start_timer` twice. So I need to prevent it.

I'll change the signature of `on_resume` and `on_pause` so both will get this flag as parameter and not from the outer scope.